src/txn/txn_ckpt.c

```diff
@@ -956,6 +962,31 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
 
     __checkpoint_verbose_track(session, "sync completed");
 
+    /* If the history store file exists on disk, update its statistic. */
+    WT_ERR(__wt_metadata_search(session, WT_METAFILE_URI, &config));
+    WT_ERR(__wt_metadata_cursor_open(session, NULL, &metac));
+    metac->set_key(metac, WT_HS_URI);
+    WT_ERR_NOTFOUND_OK(metac->search(metac), true);
+
+    if (ret != WT_NOTFOUND) {
+        WT_ERR(__wt_fs_exist(session, WT_HS_FILE, &hs_exists));
+
+        WT_ERR(__wt_block_manager_named_size(session, WT_HS_FILE, &hs_size));
+        WT_STAT_CONN_SET(session, cache_hs_ondisk, hs_size);
+    }
+
+    WT_RET(__wt_curhs_open(session, NULL, &hs_cursor));
```

Comment:
As we already have a `hs_dhandle`, you can directly get the btree from it instead of opening a cursor.